### ISSUE TITLE
Review GRIN listing on Bisq

### ISSUE THREAD

**Issue by lavinger** [bisq-network/growth#205]
Grin is a relatively new cryptocurrency taking advantage of mimblewimble, which is a protocol focused on privacy.

As I understand it, trading Grin is haulted at the moment because sending/receiving caused confusion among some users. However, Grin already uses a new standard way of sending coins: slatepack addresses. This eases the process and also unifies it.

I suggest that Grin is allowed to be traded on Bisq once again. The sending/receiving process has been improved, users should be more familiar with the coin now.

I also suggest we add some simple UI for Grin trading. For example, a simple pop up window refering them to Grin documentation.

As Grin focuses on privacy just as Bisq does, I think it could be very valuable. Monero, another privacy coin, is also doing very good on Bisq. Just imagine becoming one of the main trading exchanges for Grin.

**Comment by m52go** [comment]
@chimp1984 any thoughts here? I don't follow Grin very much.

This was the biggest discussion thread on it from what I remember:
https://bisq.community/t/we-need-to-remove-grin/7013/20

**Comment by chimp1984** [comment]
Is there a way to verify the Grin transfer, so that the sender of Grin can proof to the mediator or arbitrator that he sent it? Somthing like the private tx key on Monero?
Basically I am in favor to add it again if the earlier problems are solved.

**Comment by klavinger** [comment]
> Is there a way to verify the Grin transfer, so that the sender of Grin can proof to the mediator or arbitrator that he sent it?

Yes, there is. Anyone with synced blockchain can verify that a transaction took place.
[Grin Documentation - Proof of Payment](https://docs.grin.mw/getting-started/wallet-handbook/#proof)

**Comment by chimp1984** [comment]
> > Is there a way to verify the Grin transfer, so that the sender of Grin can proof to the mediator or arbitrator that he sent it?
> 
> Yes, there is. Anyone with synced blockchain can verify that a transaction took place.
> [Grin Documentation - Proof of Payment](https://docs.grin.mw/getting-started/wallet-handbook/#proof)
> 
> EDIT: In that Bisq forum thread posted by @m52go above, someone suggest a website that can also provide proof of payments, if syncing Grin blockchain should be an issue for mediators. However, it is quite an old post and I don't have any experience with that.
> 
> Grin blockchain is just shy of 1.9 GB atm (at least for me, there might be some variations).

Thanks. Would be great if there is a web page to do the verification so the mediators do not need to install it themselfes.

**Comment by bluimes** [comment]
Bump this issue for listing purposes. We really want to make this happen. Can you guys share your latest concerns with us? 

In agreement with Manfred I quote our previous convo: 

`Hi Manfred,

Thanks for your comment. The github issue is still open under https://github.com/bisq-network/growth/issues/205. In the previous issue the discussion stopped. https://bisq.community/t/we-need-to-remove-grin/7013/15
The last concrete question was whether there was a way to verify the payment. You can do this via https://grinscan.net

Our community member @ MCM.mike started extended testing with slatepacks. https://forum.grin.mw/t/grinnode-live-winter-2020-bug-bash-challenge-and-free-grin-btc-prizes-giveaway/8060/13. He has also indicated that he wants to continue to tackle the github issue.

Are there any other requirements from Bisq in order to start this listing? Can you ask your devs and community what other issues should be addressed?

Ultimately, I sincerely believe that the future of privacy lies with DEX.`

**Comment by MCM-Mike** [comment]
> Yes, there is. Anyone with synced blockchain can verify that a transaction took place.
> Grin Documentation - Proof of Payment
> 
> EDIT: In that Bisq forum thread posted by @m52go above, someone suggest a website that can also provide proof of payments, if syncing Grin blockchain should be an issue for mediators. However, it is quite an old post and I don't have any experience with that.

Just a short update on this topic: 

Yes we from [Grinnode.live](https://grinnode.live/) are still working on a solution for this problem. 
At the moment we are developing a public API which anyone can query, when activated and use it to verify of a payment. This can be useful when you are a `mediator` or `arbitrator` 
It can be activated via GRIN and has a standard `JSON API` to query if a transaction did take place and we might also provide a front end for it if required.

**Comment by marekyggdrasil** [comment]
Hi all! The alpha version of our proof-verification API is being tested right now. As @MCM-Mike says, it is a JSON api that verifies payment proofs generated by the wallet. Users can charge their account with Grin balance by providing payment proof to grinnode.live wallet. Every response from the API is signed with `ed25519` key which also is grinnode.live wallet. For additional security you can enable signature verification and our API will only process the requests signed with your wallet key.

**Comment by MCM-Mike** [comment]
We have published a first beta version of the **GRIN payment-proof** and made a public forum post here: https://forum.grin.mw/t/early-beta-version-of-grinnode-lives-public-transaction-proof-verification-service-is-online/9080 

For issues and requests please use the corresponding [Github](https://github.com/Grinnode-live/payment-proof-verifier) account mentioned in the forum post.

**Comment by newjack777** [comment]
I'd like to bump this issue and provide further assistance to the BISQ developers with GRiN re-listing as the necessary tools and protocols have since been developed so to address the initial barriers to BISQ user convenience and accounting.

**Comment by MCM-Mike** [comment]
Thank you @newjack777 , if anyone from BISQ has a question regarding GRIN or the payment-proof please reach out to me or anyone from the GRIN community.

**Comment by chimp1984** [comment]
@jmacxx Would you have bandwidth to cover that?

**Comment by jmacxx** [comment]
I propose to take the following action items in order to approve re-listing GRIN on Bisq.

- [ ] Install a [wallet](https://github.com/mimblewimble/grin-wallet/releases) & obtain a small amount of GRIN.
- [ ] Modify my local Bisq build to temporarily re-enable GRIN.
- [ ] Research the [slatepack address format](https://github.com/mimblewimble/grin-rfcs/blob/master/text/0015-slatepack.md#slatepackaddress), modify Bisq to use it in the altcoin account instead of grinbox format.
- [ ] Do a test self-trade (Alice/Bob) on Bisq.
- [ ] Have Bob (BTC seller) open mediation.
- [ ] Have the mediator check Alice's GRIN payment proof by running a [python script](https://github.com/Grinnode-live/payment-proof-verifier).
- [ ] Update https://bisq.wiki/ with instructions for Bisq users & mediators on performing GRIN tx proofs.
- [ ] Discuss the required GRIN tx validation procedure with @bisq-network/mediators 
- [ ] Make a Bisq PR to implement slatepack validation.
- [ ] Coordinate with @ripcurlx to remove the asset ban from Bisq.

**Comment by wiz** [comment]
IMO there are too many altcoins on Bisq already, personally I would prefer to see Bisq become Bitcoin Only instead of listing more...

**Comment by cekickafa** [comment]
> 
> 
> IMO there are too many altcoins on Bisq already, personally I would prefer to see Bisq become Bitcoin Only instead of listing more...

Define Bitcoin.

**Comment by wiz** [comment]
@chimp1984 aren't new listings on hold as per your last DAO proposal? if so then I think listing GRIN would require a DAO vote to approve it.

**Comment by chimp1984** [comment]
The stop on add altcoins has been motivated due the flood of altcoins which did not represent value to Bisq (most never got trades, many had very scammy attitude).

I think Grin does not fall at all under that categories and has a lot of value as privacy coin and much in common with Bisq's ethos. 

Bisq does support altcoin trading and 80-90% of volume comes from XMR trades. Bisq as Fiat only exchange (and Fiat is for sure the worst shitcoin) would not be sustainable with the limited revenue from Fiat trade volume.

If there is no consenus about supporting to add Grin we can make a DAO voting. But I guess we can signal also here by up/down vote and only if still not clear we can go for the vote route if that is ok.

**Comment by 2ro** [comment]
Would love to see GRIN re-added on bisq

**Comment by marekyggdrasil** [comment]
> * Have the mediator check Alice's GRIN payment proof by running a [python script](https://github.com/Grinnode-live/payment-proof-verifier).

I remain available to help anyone willing to try the [grinnode.live payment proof verification API](https://forum.grin.mw/t/early-beta-version-of-grinnode-lives-public-transaction-proof-verification-service-is-online/9080). If you open my GitHub profile, you will find ways to reach me.

**Comment by MCM-Mike** [comment]
>  Install a wallet & obtain a small amount of GRIN.

I am also available for rust wallet question in regards to the usability or any other question which I can relay to the correct person/team.

**Comment by MwithM** [comment]
While listings at Bisq have been _on hold_, USDT(erc20), LBTC and RBTC have been added. I don't see a reason not to add Grin, or remove this listing hold entirely. Bisq needs more stablecoins (erc20 fees are too expensive) and to add top10 marketcap shitcoins.
Bisq has scams like ARS and VES listed. Just let the user decide if they want to trade or not, and make it friendly to remove the ones that they don't want to deal with.
Just add a listing fee which is high ennough to make it profitable for the DAO to review the listings, and reduce spam. Those who pay for a listing could ask for a reimbursement if their alt gets enough trading fees for the DAO.

**Comment by chimp1984** [comment]
Yes the DAO listing fee carries those functionalities already. Initial fee is not really high and after that if the coin is traded there is no fee required to keep it listed.
The main readon for the hold on listing was to avoid effort for listing requests (PRs) which required dev involvements and the devs had no motivation to spend time on some random shitcoin and get spammed the pull request list.
With Grin there are definitely devs who consider it a very valuable altcoin and are willing to put effort into it.

**Comment by vironelo** [comment]
I'm against Grin listing:

- Centralized mining 84.4% of known hashrate from one pool
- Inflation bug and 51% attack
- Dying project without active developers
- The community too small
- Low volume at the existing exchanges
- Many technical issues with Grin transactions

Sources:
- https://miningpoolstats.stream/grin
- https://forum.grin.mw/t/resolved-d-o-n-o-t-u-s-e-g-r-i-n-g-r-i-n-w-a-l-l-e-t-g-r-i-n-i-r-o-n-b-e-l-l-y-u-n-t-i-l-f-u-r-t-h-e-r-n-o-t-i-c-e/8620
- https://github.com/mimblewimble/grin/graphs/commit-activity
- https://www.gate.io/trade/GRIN_BTC

**Comment by jasotimur** [comment]
> I'm against Grin listing:
> 
> * Centralized mining 84.4% of known hashrate from one pool
> * Inflation bug and 51% attack
> * Dying project without active developers
> * The community too small
> * Low volume at the existing exchanges
> * Many technical issues with Grin transactions
> 
> Sources:
> 
> * https://miningpoolstats.stream/grin
> * https://forum.grin.mw/t/resolved-d-o-n-o-t-u-s-e-g-r-i-n-g-r-i-n-w-a-l-l-e-t-g-r-i-n-i-r-o-n-b-e-l-l-y-u-n-t-i-l-f-u-r-t-h-e-r-n-o-t-i-c-e/8620
> * https://github.com/mimblewimble/grin/graphs/commit-activity
> * https://www.gate.io/trade/GRIN_BTC

you opened a new github acc for this?) @wiz

**Comment by wiz** [comment]
> While listings at Bisq have been on hold, USDT(erc20), LBTC and RBTC have been added.

I think you guys are missing my point. Since new listings are on hold, new listings must be approved by the DAO. The assets you mentioned were approved by the DAO in the following proposals and GRIN must do the same process as any other new listing.

https://github.com/bisq-network/proposals/issues/169

https://github.com/bisq-network/proposals/issues/170

**Comment by 2ro** [comment]
> GRIN must do the same process as any other new listing.
GRIN is not a new listing.

**Comment by cekickafa** [comment]
Bisq volume is about 250k atm. i think Grin will add more liquidity to this exchange.its a win-win for both ecosystem.

**Comment by minexpert** [comment]
@jasotimur Nice points. I will give some answer. 

- There are currently 3600 miners and approximately $10 million of mining equipment is required to provide this network power. Grin has this. It's just that most of the miners are from China, so they use Chinese origin pools and we are working to fix this issue.
- Inflation bug has been resolved and Grin network power is increasing day by day. So this is unlikely.
- Definitely not a dying project. All the developers have done their work and the hard-forks are complete. They left because there was nothing more to do. Currently, Grin has enough budget for developers.
- Grin is an emerging community. 
- The community is growing up everyday.
- There is good volume in Gate and Tradeogre for Grin's current position. Months ago, Grin had $2-3 million in volume on Gate.
- Technical problem? People who do not know Grin may have problems, but for those who do, there is no problem. Just try sending Grin via tradeogre. You will see how practical and hassle-free it is.

Another points; 
1. Grin's network power is not even available in cryptocurrencies that are worth 10 times more than Grin.
2. Grin's daily transfer count is even higher than cryptocurrencies that are worth 10 times more. For example, you can compare it on coinmetrics with Verge coin.
3. Grin's daily transfer count is higher than any of Monero and Dash's early years.
4. Grin is a completely decentralized and fair cryptocurrency. It should be added to the stock market not for an interest, but because it is so.
5. The exclusion of Grin today may cause your losses. Because the comminity doesn't forget that. Today Grin may need you, but tomorrow you may need Grin.

**Comment by chimp1984** [comment]
I just made a DAO proposal for it: 
https://github.com/bisq-network/proposals/issues/350

**Comment by jaw709** [comment]
I want to add amidst all the noise of pointless coins with huge premine, copy and pasted code, large front end distributions, heavy node running requirements, grin offers something unique:  scalable self-validation keeping pace with bandwidth AND default privacy on baselayer with confidential amounts.

We recently had a bright developerbcome on board who is implementing coinswap which will help solve the "linkability" issue which caused so many to turn away early on. Grin has the potential to become a beautiful balance of premier scalability and superior privacy to most garbage coins, and it would dishonor the mission of cryptographically sound, decentralized money if we lost this plot.

It has had a tumultuous past but we are primed to do something great here, I hope this comes along for this journey.

**Comment by chimp1984** [comment]
I for my part am totally aware of the importance of Grin. It is beside Monero the most cypherpunk aliged coin IMO. 
I also think that privacy coins play a crucial role to help to mitigate Bitcoins weak privacy. Swapping Bitcoin to privacy coins is one important tool to ensure Bitcoin will not end up in a total surveillance nightmare.

**Comment by jasotimur** [comment]
Bisq+Grin=Fits good.  it is already listed ,let the implement begin. 
No need too much drama.
Hit the iron while it is hot.

**Comment by MwithM** [comment]
@cekickafa I'm skeptical about the volume that Grin could generate. But in doubt, let's list it.
@vironelo I think I'll never use Grin. Maybe I should give that a deeper thought, but there are much worse coins listed at Bisq. Listing a coin should not be equal to endorse it.
@wiz So, instead of replying to those who want to get their shitcoin listed that Bisq listings are on hold, should I reply them that new listings should be approved by the DAO? It's not a bad policy. At least it's a policy, much better than the plain hold. I still think that Bisq should take profit just by listing coins, even if they're not traded afterwards.

**Comment by bladedoyle** [comment]
> * Centralized mining 84.4% of known hashrate from one pool
> * Inflation bug and 51% attack
> * Dying project without active developers
> * The community too small
> * Low volume at the existing exchanges
> * Many technical issues with Grin transactions

The mining centralization is an issue, but not as much as you suggest.  the 84% you see is for pools, that one pool has 84% of the pool hashrate - but only 68% of the network hashrate.

Every crypto project has the risk of bugs.  Its true that Grin had an inflation bug but it was identified and fixed almost immediately.  Its true there was successful 51% attack against an exchange using grin.  This is not uncommon.  Although ASIC have centralized mining for Grin, it has minimized the risk of 51% attacks going forward.

The community is currently small.  But they have both a development fund and a community project fund.  The original devs were very exclusive at first but most of them left so there is space to fill and funds to help fill it.  Being part of Bisq can help increase awareness and use of Grin.

ya, p2p interactive transactions are hard to fit into existing models.

**Comment by jmacxx** [comment]
Great to see such engagement on this thread!  I've started work on [this task list](https://github.com/bisq-network/growth/issues/205#issuecomment-962480146) ahead of the DAO vote and will post updates here to keep you informed.  Thanks for the offers of assistance @marekyggdrasil and @MCM-Mike, I'll be reaching out to you on keybase soon.  Also to @2ro to [buy some GRIN](https://slatepacks.com/post/9).

**Comment by persepolise** [comment]
i hope grin can earn popularity and be in the top i would like to see grin being traded at a dex

**Comment by jmacxx** [comment]
Most of the [changes for re-listing](https://github.com/bisq-network/growth/issues/205#issuecomment-962480146) have been done and :crossed_fingers: should make it into the next release about the same time as the DAO vote results (2021-12-01).  If the DAO vote passes then it would just be a configuration switch to enable trading.

**Comment by jmacxx** [comment]
- [x] https://github.com/bisq-network/proposals/issues/350 was approved by Bisq DAO vote at Cycle 30 / block height 712137.
- [ ] GRIN trading requires Bisq release v1.8.0 which is in the pipeline.. coming soon<sup>(TM)</sup>

**Comment by jmacxx** [comment]
Bisq v1.8.0 is available for download, and GRIN market is activated.  https://github.com/bisq-network/bisq/releases/

Hope to see some more offer makers there!

![image](https://user-images.githubusercontent.com/47253594/145630858-ad86b04d-11f4-4d3b-9ab2-3581c6a30a64.png)

**Comment by MCM-Mike** [comment]
Thank you BisQ for listing GRIN again. 
It was a joint effort between us, the [GRIN community](https://forum.grin.mw/t/bisq-exchange-dao-vote-on-supporting-grin-again/9352/15?u=mcm-mike) and you BisQ.

**Comment by marthaabel** [comment]
I was convinced to invest here by a friend I met on Tinder, Her name was Annie ( now I know she could be anyone else). At first everything was going on well until I requested for withdrawal, that’s when the nightmare started… I had to put in more money before I could withdraw. Then came the unending taxes; I was fined and taxed for various reasons. There was always a reason why I couldn’t take out my money. I was tired of everything and began looking for help. Since all the payments I made were through cryptocurrencies, my local police could not help me. Luckily I finally got help from the authorities I contacted via email at zattrecoverypro1 @ gmail com. Wonderfully, after providing my details, it took them barely a week to trace and retrieve my funds. Get help from them if you are in a similar.

**Comment by Nelson-gr** [comment]
I’m pleased to recommend this genuine funds recovery service, he is the best and reliable fund recovery agency to help you get back your lost, scammed, stolen cryptocurrencies or payment you made by bank transfer back. contact (hackerthrone1@ gmail .com) They are the real deal as I was able to recover my stolen USDT back after days of being depressed. I'm grateful.

**Comment by Anabelle42** [comment]
There are a lot of untrue recommendations and its hard to tell who is legit. If you have lost money to scam contact (zattrecoverypro1 ⓐ Gmail Dot Com) they will surely help you out. Took me long to find them

**Comment by janeljoose** [comment]
**Losing money can be terrible. It can constantly cause despair and misery,which is precisely what occurred to me,to say the least. Everything seemed authentic until I realized I had been duped after investing $150,000 in cryptocurrency in an online investment racket.After some time I came across a review about ULTIMATE RECOVERY FIRM and how they assist victims of fraud in recovering their lost funds, I had been depressed for days. I was hesitant at first but then decided to give them a try. After several procedures I was able to get my money back within a couple of days. Please be cautious and take care to understand the investments you are making. Anyone else who is a victim can contact ULTIMATE RECOVERY FIRM through email address Email: ultimate_recovery _firm_@_gmail._com WhatsApp: +1 -- 786-- 432 --2178....**
